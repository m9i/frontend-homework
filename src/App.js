import "./stylesheets/index.scss";

import DropdownLink from "./components/DropdownLink/DropdownLink";
import DropdownItem from "./components/DropdownItem/DropdownItem";

const App = () => (
  <div>
    <nav className="nav">
      <DropdownLink />
      <DropdownItem />
    </nav>
  </div>
);

export default App;
