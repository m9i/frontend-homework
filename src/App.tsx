import './stylesheets/index.scss';
import DropdownLink from './components/DropdownLink/DropdownLink';
import Dropdown from './components/Dropdown/Dropdown';
import {connect} from 'react-redux';
import React, {useEffect} from 'react';
import {toggleDropdownAction} from './store/actions/toggleDropdownAction';
import {IAppProps} from './interfaces/IAppProps';
import {IState} from './interfaces/store/IState';

const App: React.FC<IAppProps> = ({showDropdown, hideDropdown}: IAppProps) => {
    const checkIfOutside = (event: any): boolean => {
        let clickedElement: any = event.target;

        while (clickedElement) {
            let navLinkOrDropDownClicked =
                clickedElement.classList.contains('nav__list') ||
                clickedElement.classList.contains('nav__link');
            if (navLinkOrDropDownClicked) {
                return false;
            }
            clickedElement = clickedElement.parentElement;
        }
        return true;
    };
    useEffect(() => {
        const onClick = (mouseEvent: any) => {
            if (checkIfOutside(mouseEvent)) {
                hideDropdown();
            }
        };
        window.addEventListener('click', (e) => onClick(e));
        return () => {
            window.removeEventListener('resize', onClick, false);
        };
    }, [hideDropdown]);
    return (
        <div>
            <nav className="nav">
                <DropdownLink/>
                {showDropdown && <Dropdown/>}
            </nav>
        </div>
    );
};

const mapStateToProps = (state: IState) => ({
    showDropdown: state.dropdown.showDropdown,
});

const mapDispatchToProps = (dispatch: any) => ({
    hideDropdown: () => dispatch(toggleDropdownAction(false)),
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
