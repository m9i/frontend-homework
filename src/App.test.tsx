import React from 'react';
import App from './App';
import {shallow} from 'enzyme';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';

describe('rendering components', () => {
    const mockStore = configureMockStore([]);
    let store: any;

    beforeEach(() => {
        store = mockStore({
            company: {
                selectedCompany: {value: 'mock-name'},
            },
            dropdown: {
                showDropdown: false,
            },
        });
    });

    it('Renders App component without crashing', () => {
        const AppComponent = shallow(
            <Provider store={store}>
                <App/>
            </Provider>
        ).dive();
        expect(AppComponent);
    });

    it('snapshot', () => {
        const AppComponent = shallow(
            <Provider store={store}>
                <App/>
            </Provider>
        ).dive();
        expect(AppComponent).toMatchSnapshot();
    });
});

describe('App', () => {
});
