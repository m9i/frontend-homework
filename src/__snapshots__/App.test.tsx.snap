// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering components snapshot 1`] = `<Connect(App) />`;
